services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/IntersectionController.java

```diff
@@ -0,0 +1,89 @@
+package us.dot.its.jpo.ode.api.controllers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import us.dot.its.jpo.ode.api.accessors.map.ProcessedMapRepository;
+import us.dot.its.jpo.ode.api.models.IntersectionReferenceData;
+
+@RestController
+@ConditionalOnProperty(name = "enable.api", havingValue = "true", matchIfMissing = false)
+@ApiResponses(value = {
+        @ApiResponse(responseCode = "401", description = "Unauthorized"),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+})
+public class IntersectionController {
+
+    @Autowired
+    ProcessedMapRepository processedMapRepo;
+
+    // TODO: Limit response by intersections which organization the user is
```

Comment:
question: should this be done before merging? It's a security issue for multiple organization deployments if not right?

Reply:
This change has been made - returned intersections are now filtered by organization/role